test/functionalTest/cases/3189_oneshot_subscription/3189_oneshot_subscription.test

```diff
@@ -0,0 +1,313 @@
+      "attrs": [
+        "pressure"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [
+      "temperature"
+    ]
+  },
+  "status" : "oneshot"
+}'
+
+orionCurl --url /v2/subscriptions --payload "$payload"
+subId=$(echo "$_responseHeaders" | grep Location | awk -F/ '{ print $4 }' | tr -d "\r\n")
+echo
+echo
+
+
+echo '03. PATCH /v2/entities/Room1/attrs, to update the value of pressure attribute.'
+echo '=============================================================================='
+payload='{
+  "temperature": {
+    "value": 24,
+    "type": "Float"
+  },
+  "pressure": {
+    "value": 730,
+    "type": "Float"
+  }
+}'
+orionCurl --url /v2/entities/Room1/attrs --payload "$payload" -X PATCH
+echo
+echo
+
+
+echo '04. Dump and reset accumulator, see one notification.'
+echo '====================================================='
+accumulatorDump
+accumulatorReset
+echo
+echo
+
+
+echo '05. GET /v2/subscriptions/subscription_id, to check the status.'
+echo '==============================================================='
+orionCurl --url /v2/subscriptions/${subId}
+echo
+echo
+
+
+echo '06. PATCH /v2/subscriptions/subscription_id, to update status with oneshot.'
+echo '==========================================================================='
+payload='{
+	"status" : "oneshot"
+}'
+orionCurl --url /v2/subscriptions/${subId} --payload "$payload" -X PATCH
+echo
+echo
+
+
+echo '07. PATCH /v2/entities/Room1/attrs, to update the value of pressure attribute.'
+echo '=============================================================================='
+payload='{
+  "temperature": {
+    "value": 25,
+    "type": "Float"
+  },
+  "pressure": {
+    "value": 740,
+    "type": "Float"
+  }
+}'
+orionCurl --url /v2/entities/Room1/attrs --payload "$payload" -X PATCH
+echo
+echo
+
+
+echo '08. Dump and reset accumulator, see one notification.'
+echo '====================================================='
+accumulatorDump
+accumulatorReset
+echo
+echo
+
+
+--REGEXPECT--
+01. POST /v2/entities, to create Room1 with temperature and pressure.
+=====================================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/Room1?type=Room
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+02. POST /v2/subscriptions, to create subscription with status oneshot.
+=======================================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+03. PATCH /v2/entities/Room1/attrs, to update the value of pressure attribute.
+==============================================================================
+HTTP/1.1 204 No Content
+Content-Length: 0
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+04. Dump and reset accumulator, see one notification.
```

Comment:
Good to see just one notification here. It works as expected :)

However, how did you avoid to trigger initial notification? Looking to the code, it seems it doesn't include any modification to achieve that... could you elaborate on it, please?

Reply:
There is a [check](https://github.com/telefonicaid/fiware-orion/blob/master/src/lib/mongoBackend/MongoGlobal.cpp#L2499) in "**static BSONArray processConditionVector**" method in **MongoGlobal.cpp** file that initial notification is sent to only those context Consumer which have status active. In this case, status in oneshot and so initial notification will never be sent to the context consumer.

Reply:
Thanks for the clarification. NTC